### PR TITLE
Update `avoid-leaking-state-in-ember-objects` rule to apply to mixins

### LINKED ISSUE #219: Mixins should be included in avoid-leaking-state rule
Doesn't look like Mixins are checked to see if they pass the avoid-leaking-state rule. Should they be checked?

### PR DESCRIPTION
**Pull request by jaydgruber**
## What

- applies `avoid-leaking-state` rule to Ember Mixins as well
- updates the docfile a bit with more references & to mention native classes

### Why

- fixes #219

### CODE REVIEW

File: docs/rules/avoid-leaking-state-in-ember-objects.md [hunk 1]
```diff
@@ -20,6 +20,12 @@ export default Foo.extend({
 });
 ```
 
+```javascript
+export default Mixin.create({
```
Comment (bmish) [inline docs/rules/avoid-leaking-state-in-ember-objects.md:24]: Can you add the mixin import to this example for completeness?

File: tests/lib/rules/avoid-leaking-state-in-ember-objects.js [hunk 2]
```diff
@@ -66,6 +66,9 @@ eslintTester.run('avoid-leaking-state-in-ember-objects', rule, {
     'export default Foo.extend({ foo: condition ? "foo" : "bar" });',
     'export default Foo.extend({ foo: "foo" && "bar" });',
     'export default Foo.extend({ foo: "foo" || "bar" });',
+    'export default Mixin.create({});',
+    'export default Mixin.create({ harmlessProp: "foo" });',
+    'export default Mixin.create(NestedMixin, {});',
   ],
```
Comment (bmish) [inline tests/lib/rules/avoid-leaking-state-in-ember-objects.js:72]: Can you add some tests that native classes are ignored for mixins and Ember objects?

**Comment by boris-petrov** [comment]
I believe this is missing two cases:

```javascript
Mixin.create({
  // both get the warning and both shouldn't
  foo: undefined as string | undefined,
  bar: undefined as any,
})

And service-injections:

```javascript
Mixin.create({
  store: (inject() as unknown) as Store,
})
```

All of these get a warning but all of them are fine I believe.

**Comment by bmish** [comment]
@boris-petrov can you create an issue?

**Pull request merged**